app/policies/ticket_desk_policy.rb

```diff
@@ -0,0 +1,31 @@
+class TicketDeskPolicy < ApplicationPolicy
+  class Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def create?
+      authorization?
+    end
+
+    def resolve?
+      if user.admin?
+        scope.all
+      else
+        scope.where(published: true)
+      end
+    end
+
+    def update?
+      user_signed_in? && user.admin?
```

Comment:
If User would be not be signed in the value `user` would be `nil` and it would fail anyway.
Pundit is a tool for Authorization, not Authentication - policies themselves should not care whenever a user is logged in or not, that's what devise is for.